src/python/pants/pantsd/service/scheduler_service.py

```diff
@@ -196,7 +196,7 @@ def _prefork_loop(self, session, options):
         continue
     return target_roots
 
-  def _prefork_body(self, session, options):
+  def _prefork_body(self, session, options, options_bootstrapper):
```

Comment:
There's something a bit jarring about passing around an `options` along with the `options_bootstrapper` that created it, since, naively, the role of the latter is create the former, so one might wonder why you still need the bootstrapper around, or, if you do, why you can't generate `options` from it. Can you clarify in a comment somewhere? Or at least on the review, for my own edification?

Reply:
Getting `Options` from an `OptionsBootstrapper` requires (re-)running the `OptionsInitializer` to collect known scopes and call `get_full_options`... the in-engine options parsing current re-does that work if requested. But as we port more code to v2, we should be able to push `TargetRoots` calculation and other options-consuming prerequisite code into the engine and make that the only instance of options parsing in a v2-only run.

I _think_ that this `options` instance is actually only the global options and could be renamed to make that more obvious, but assuming this CI run goes green, I'd rather due that followup in #6880 instead.

Reply:
Ah, that makes sense. Definitely don't rename it in this change.